src/components/dms/ActionsWrapper.web.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react'
+import {NativeSyntheticEvent, StyleSheet, View} from 'react-native'
+import {ChatBskyConvoDefs} from '@atproto-labs/api'
+
+import {atoms as a} from '#/alf'
+import {MessageMenu} from '#/components/dms/MessageMenu'
+import {useMenuControl} from '#/components/Menu'
+
+export function ActionsWrapper({
+  message,
+  isFromSelf,
+  children,
+}: {
+  message: ChatBskyConvoDefs.MessageView
+  isFromSelf: boolean
+  children: React.ReactNode
+}) {
+  const menuControl = useMenuControl()
+  const viewRef = React.useRef(null)
+
+  const [showActions, setShowActions] = React.useState(false)
+
+  const onMouseEnter = React.useCallback(() => {
+    setShowActions(true)
+  }, [])
+
+  const onMouseLeave = React.useCallback(() => {
+    setShowActions(false)
+  }, [])
+
+  // We need to handle the `onFocus` separately because we want to know if there is a related target (the element
+  // that is losing focus). If there isn't that means the focus is coming from a dropdown that is now closed.
+  // TODO what is the type for this?
+  const onFocus = React.useCallback((e: NativeSyntheticEvent<any>) => {
```

Comment:
@mozzius do you know what the type is for this?

Reply:
(whatever it is, needs to be wrapped in `NativeSyntheticEvent` since `View` returns it in `event.nativeEvent`)